### ISSUE TITLE
Problems getting TAP 0.2.1 working on docker-kind unmanaged-cluster

### ISSUE THREAD

**Issue by Analect** [vrabbi/tap-oss#4]
Running on kind unmanaged cluster (TCE v 0.10.rc4).  Installing vrabbi/top-oss (v0.2.1 of repo and package).
While contour eventually resolves itself. knative and kpack do not.

```
~/dev/tce/tap-oss$ kubectl get pkgi -n tap-oss
NAME                     PACKAGE NAME                     PACKAGE VERSION   DESCRIPTION                                                            AGE
cartographer             cartographer.tap.oss             0.2.1             Reconcile succeeded                                                    23m
cert-injection-webhook   cert-injection-webhook.tap.oss   0.1.0             Reconcile succeeded                                                    23m
cert-manager             cert-manager.tap.oss             1.6.1             Reconcile succeeded                                                    23m
contour                  contour.tap.oss                  1.19.1            Reconcile failed: Error (see .status.usefulErrorMessage for details)   23m
dev-ns-preperation       dev-ns-preperation.tap.oss       0.2.1             Reconcile succeeded                                                    23m
flux-source-controller   flux-source-controller.tap.oss   0.21.1            Reconcile succeeded                                                    23m
knative                  knative-serving.tap.oss          1.0.0             Reconcile failed: Error (see .status.usefulErrorMessage for details)   23m
kpack                    kpack.tap.oss                    0.5.0             Reconcile failed: Error (see .status.usefulErrorMessage for details)   23m
kpack-config             kpack-config.tap.oss             0.5.2             Reconcile succeeded                                                    23m
ootb-supply-chains       ootb-supply-chains.tap.oss       0.2.1             Reconcile succeeded                                                    23m
service-bindings         service-bindings.tap.oss         0.6.0             Reconcile succeeded                                                    23m
tap                      tap-install.tap.oss              0.2.1             Reconcile succeeded                                                    23m
tekton-pipelines         tekton.tap.oss                   0.32.1            Reconcile succeeded                                                    23m


/dev/tce/tap-oss$ kubectl get pkgi -n tap-oss
NAME                     PACKAGE NAME                     PACKAGE VERSION   DESCRIPTION                                                            AGE
cartographer             cartographer.tap.oss             0.2.1             Reconcile succeeded                                                    25m
cert-injection-webhook   cert-injection-webhook.tap.oss   0.1.0             Reconcile succeeded                                                    25m
cert-manager             cert-manager.tap.oss             1.6.1             Reconcile succeeded                                                    25m
contour                  contour.tap.oss                  1.19.1            Reconcile succeeded                                                    25m
dev-ns-preperation       dev-ns-preperation.tap.oss       0.2.1             Reconcile succeeded                                                    25m
flux-source-controller   flux-source-controller.tap.oss   0.21.1            Reconcile succeeded                                                    25m
knative                  knative-serving.tap.oss          1.0.0             Reconcile failed: Error (see .status.usefulErrorMessage for details)   25m
kpack                    kpack.tap.oss                    0.5.0             Reconcile failed: Error (see .status.usefulErrorMessage for details)   25m
kpack-config             kpack-config.tap.oss             0.5.2             Reconcile succeeded                                                    25m
ootb-supply-chains       ootb-supply-chains.tap.oss       0.2.1             Reconcile succeeded                                                    25m
service-bindings         service-bindings.tap.oss         0.6.0             Reconcile succeeded                                                    25m
tap                      tap-install.tap.oss              0.2.1             Reconcile succeeded                                                    25m
tekton-pipelines         tekton.tap.oss                   0.32.1            Reconciling                                                            25m

```
Here are some possible relevant logs. 
kpack-webook pod is failing to pull an image.

```
kpack-webhook-xxxx
Failed to pull image "projects.registry.vmware.com/tce/kpack@sha256:9b62ad735b7df4577e3e8b06e5193d34434b11bffc5baf96ac7a9707971bb4e1": rpc error: code = FailedPrecondition desc = failed to pull and unpack image "projects.registry.vmware.com/tce/kpack@sha256:9b62ad735b7df4577e3e8b06e5193d34434b11bffc5baf96ac7a9707971bb4e1": failed commit on ref "layer-sha256:7fd31a3e1ee826fb3b4cd97516ebb2df8b0b722e9d2b84b2742ed1856fd79782": "layer-sha256:7fd31a3e1ee826fb3b4cd97516ebb2df8b0b722e9d2b84b2742ed1856fd79782" failed size validation: 4194304 != 24851776: failed precondition
```
Also, this is from running `kubectl get pkgi -n tap-oss kpack -o yaml`.  Not sure there's any incrimental information here.
```
status:
  conditions:
  - message: Error (see .status.usefulErrorMessage for details)
    status: "True"
    type: ReconcileFailed
  friendlyDescription: 'Reconcile failed: Error (see .status.usefulErrorMessage for
    details)'
  observedGeneration: 1
  usefulErrorMessage: |-
    kapp: Error: waiting on reconcile deployment/kpack-webhook (apps/v1) namespace: kpack:
      Finished unsuccessfully (Deployment is not progressing: ProgressDeadlineExceeded (message: ReplicaSet "kpack-webhook-d6b549b79" has timed out progressing.))
  version: 0.5.0
```
Output below is from running `kubectl get pkgi -n tap-oss knative -o yaml`.  This seems to result from the activator (logs below) unable to get running successfully.

```
status:
  conditions:
  - message: Error (see .status.usefulErrorMessage for details)
    status: "True"
    type: ReconcileFailed
  friendlyDescription: 'Reconcile failed: Error (see .status.usefulErrorMessage for
    details)'
  observedGeneration: 1
  usefulErrorMessage: |-
    kapp: Error: waiting on reconcile deployment/activator (apps/v1) namespace: knative-serving:
      Finished unsuccessfully (Deployment is not progressing: ProgressDeadlineExceeded (message: ReplicaSet "activator-b7d448954" has timed out progressing.))
  version: 1.0.0
```

Here are some logs from that activator.
```
knative-serving
activator-xxx
Liveness probe failed: HTTP probe failed with statuscode: 500
{"severity":"INFO","timestamp":"2022-02-11T20:35:52.31035659Z","caller":"logging/config.go:116","message":"Successfully created the logger."}
{"severity":"INFO","timestamp":"2022-02-11T20:35:52.310671491Z","caller":"logging/config.go:117","message":"Logging level set to: info"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:52.813982569Z","logger":"activator","caller":"activator/main.go:134","message":"Starting the knative activator","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:52.964016461Z","logger":"activator","caller":"activator/main.go:179","message":"Connecting to Autoscaler at ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:52.964145161Z","logger":"activator","caller":"profiling/server.go:64","message":"Profiling enabled: false","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:52.964900463Z","logger":"activator","caller":"websocket/connection.go:162","message":"Connecting to ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:53.077289482Z","logger":"activator","caller":"metrics/metrics_worker.go:76","message":"Flushing the existing exporter before setting up the new exporter.","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:53.077566882Z","logger":"activator","caller":"metrics/prometheus_exporter.go:51","message":"Created Prometheus exporter with config: &{knative.dev/internal/serving activator prometheus 5000000000 <nil> <nil>  false 9090 0.0.0.0}. Start the server for Prometheus exporter.","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:53.077605782Z","logger":"activator","caller":"metrics/metrics_worker.go:91","message":"Successfully updated the metrics exporter; old config: <nil>; new config &{knative.dev/internal/serving activator prometheus 5000000000 <nil> <nil>  false 9090 0.0.0.0}","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"INFO","timestamp":"2022-02-11T20:35:53.078732984Z","logger":"activator","caller":"activator/request_log.go:45","message":"Updated the request log template.","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","template":""}
{"severity":"ERROR","timestamp":"2022-02-11T20:35:55.966091698Z","logger":"activator","caller":"websocket/connection.go:145","message":"Websocket connection could not be established","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"dial tcp 52.128.23.153:8080: i/o timeout","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:145\nknative.dev/pkg/websocket.(*ManagedConnection).connect.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:226\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:211\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:399\nknative.dev/pkg/websocket.(*ManagedConnection).connect\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:223\nknative.dev/pkg/websocket.NewDurableConnection.func2\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:163"}
{"severity":"ERROR","timestamp":"2022-02-11T20:35:56.302840753Z","logger":"activator","caller":"websocket/connection.go:192","message":"Failed to send ping message to ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"connection has not yet been established","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func3\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:192"}
{"severity":"ERROR","timestamp":"2022-02-11T20:35:59.089123268Z","logger":"activator","caller":"websocket/connection.go:145","message":"Websocket connection could not be established","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"dial tcp 52.128.23.153:8080: i/o timeout","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:145\nknative.dev/pkg/websocket.(*ManagedConnection).connect.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:226\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:211\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:399\nknative.dev/pkg/websocket.(*ManagedConnection).connect\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:223\nknative.dev/pkg/websocket.NewDurableConnection.func2\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:163"}
{"severity":"ERROR","timestamp":"2022-02-11T20:35:59.636541831Z","logger":"activator","caller":"websocket/connection.go:192","message":"Failed to send ping message to ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"connection has not yet been established","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func3\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:192"}
{"severity":"WARNING","timestamp":"2022-02-11T20:35:59.849944446Z","logger":"activator","caller":"handler/healthz_handler.go:36","message":"Healthcheck failed: connection has not yet been established","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c"}
{"severity":"ERROR","timestamp":"2022-02-11T20:36:02.272950451Z","logger":"activator","caller":"websocket/connection.go:145","message":"Websocket connection could not be established","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"dial tcp 52.128.23.153:8080: i/o timeout","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:145\nknative.dev/pkg/websocket.(*ManagedConnection).connect.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:226\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:211\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:399\nknative.dev/pkg/websocket.(*ManagedConnection).connect\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:223\nknative.dev/pkg/websocket.NewDurableConnection.func2\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:163"}
{"severity":"ERROR","timestamp":"2022-02-11T20:36:02.974539713Z","logger":"activator","caller":"websocket/connection.go:192","message":"Failed to send ping message to ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"connection has not yet been established","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func3\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:192"}
{"severity":"ERROR","timestamp":"2022-02-11T20:36:05.449860817Z","logger":"activator","caller":"websocket/connection.go:145","message":"Websocket connection could not be established","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"dial tcp 52.128.23.153:8080: i/o timeout","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:145\nknative.dev/pkg/websocket.(*ManagedConnection).connect.func1\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:226\nk8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:211\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\tk8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:399\nknative.dev/pkg/websocket.(*ManagedConnection).connect\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:223\nknative.dev/pkg/websocket.NewDurableConnection.func2\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:163"}
{"severity":"ERROR","timestamp":"2022-02-11T20:36:06.302349471Z","logger":"activator","caller":"websocket/connection.go:192","message":"Failed to send ping message to ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"f1e419a","knative.dev/controller":"activator","knative.dev/pod":"activator-b7d448954-vsj6c","error":"connection has not yet been established","stacktrace":"knative.dev/pkg/websocket.NewDurableConnection.func3\n\tknative.dev/pkg@v0.0.0-20211101212339-96c0204a70dc/websocket/connection.go:192"}
Logs from 2/11/2022, 8:35:51 PM
Show timestamps
Show previous terminated container
```

Also, multiple default-domain-xx jobs failing in the knative-serving namespace.
```
knative-serving: default-domain-xxxx (many failed jobs)
W0211 21:40:20.766542       1 client_config.go:615] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
{"level":"fatal","ts":1644616832.4148066,"logger":"fallback.default-domain","caller":"default-domain/main.go:199","msg":"Error finding gateway address","error":"timed out waiting for the condition","stacktrace":"main.main\n\tknative.dev/serving/cmd/default-domain/main.go:199\nruntime.main\n\truntime/proc.go:255"}
```

**Comment by vrabbi** [comment]
as discussed on slack, this looks like a docker issue. there seems to be an issue with pulling images from docker hub within the cluster which is causing issues. once you figure that issue out if you still have any issues with the packages here in the repo please reopen this issue.
for now I am closing this issue.

thanks!

**Comment by Analect** [comment]
Just leaving a bread-crumb to a possible solution for others that might have experienced similar issues.  As per the hunch of @vrabbi , it was apparent there were problems with the DNS set-up on a local ubuntu 20.04 VM, which was having problems resolving some (but not all) docker-image pulls from dockerhub. I can't fully explain it, but here were a few steps I used to get things running again that I gleaned from various stackoverflow entries and elsewhere.

Your `/etc/resolv.conf` file (symlinked to `/run/systemd/resolve/stub-resolv.conf`) might look like this:
```
nameserver 127.0.0.53
options edns0 trust-ad
search localdomain
```
Run ` nmcli dev show | grep 'IP4.DNS'` and use this IP address to copy in as (in my case) `DNS=192.168.139.2` under `[Resolve]` in `/etc/systemd/resolv.conf`.  Run `sudo systemctl restart networking` and `service systemd-resolved restart`.

For good measure, I also added in that DNS address to a `/etc/docker/daemon.json` file (create a new one, if none pre-exists).
```
{
    "dns": ["192.168.139.2", "8.8.8.8"],
    "debug": true
}
```
Then ran: `sudo systemctl daemon-reload` and `sudo systemctl restart docker`.  This should then allow you to see that DNS Server when running `systemd-resolve --status`.  That appears to have fixed my issues in getting these great `tap-oss` resources working.
